tests/eigen_svd/sandstone_eigen_common.h

```diff
@@ -32,23 +32,28 @@ template <typename SVD, int Dim> struct EigenSVDTest
     }
 
     template <typename FP> static inline std::enable_if_t<boost::is_complex<FP>::value>
-    compare_or_fail(const FP *actual, const FP *expected, const char *name)
+    compare_or_fail(const FP *actual, const FP *expected, typename Mat::Index rows,
+                    typename Mat::Index cols, const char *name)
     {
         memcmp_or_fail(reinterpret_cast<const typename FP::value_type *>(actual),
                        reinterpret_cast<const typename FP::value_type *>(expected),
-                       2 * Dim * Dim, name);
+                       2 * rows * cols, name);
     }
 
     template <typename FP> static inline std::enable_if_t<!boost::is_complex<FP>::value>
-    compare_or_fail(const FP *actual, const FP *expected, const char *name)
+    compare_or_fail(const FP *actual, const FP *expected, typename Mat::Index rows,
+                    typename Mat::Index cols, const char *name)
     {
-        memcmp_or_fail(actual, expected, Dim * Dim, name);
+        memcmp_or_fail(actual, expected, rows * cols, name);
```

Comment:
This should be correlated to the FP size. Otherwise, it's shorter region to compare.